pystac/extensions/projection.py

```diff
@@ -65,7 +68,9 @@ class ProjectionExtension(
 
     def apply(
         self,
-        epsg: int | None,
+        *,
+        epsg: int | None = None,
+        code: str | None = None,
```

Comment:
@gadomski if we want strictly no API changes we could take away the no positional args constraint, but I think we'd want to add it in pystac v2 to make the behavior very explicit.

Reply:
Yeah I think that's fine, might break a few folks but it's an easy fix on their part.